src/NextGen/DonationForm/DataTransferObjects/DonationConfirmationReceiptViewRouteData.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Give\NextGen\DonationForm\DataTransferObjects;
+
+use function strlen;
+
+/**
+ * @unreleased
+ */
+class DonationConfirmationReceiptViewRouteData
+{
+    /**
+     * @var string|null
+     */
+    public $receiptId;
+
+    /**
+     *
+     * @unreleased
+     */
+    public static function fromRequest(array $request): DonationConfirmationReceiptViewRouteData
+    {
+        $self = new self();
+
+        $self->receiptId = isset($request['receipt-id']) && $self->isValidReceiptId(
+            $request['receipt-id']
+        ) ? $request['receipt-id'] : null;
+
+        return $self;
+    }
+
+    /**
+     * @unreleased
+     */
+    protected function isValidReceiptId(string $receiptId): bool
+    {
+        return strlen($receiptId) === 32;
```

Comment:
Here is another instance of that magic number `32`. It would probably be helpful to centralize this value with documentation.

Reply:
Resolved by adding description for now. fd7d6bc